openedx/paragon
# feat: add types for <ModalDialog> and some related components

## Description

This adds TypeScript types for:
- `ModalContext`
- `ModalDialog`
- `ModalDialog.Header`
- `ModalLayer`
- `Portal`

Before this change - no type checking in IDE, no inline docs:
![Screenshot](https://github.com/user-attachments/assets/da939222-4454-4069-9d35-dc4a48854c87)

After this change - type checking in IDE:
![Screenshot showing missing title property](https://github.com/user-attachments/assets/17b69e14-c5c7-4dbd-8a18-905fdf3bb6f7)

After this change - inline documentation of props in the IDE:
![Screenshot](https://github.com/user-attachments/assets/08fd44a3-b42c-49db-b09c-e41d800af0c8)

### Deploy Preview

TBD

### How to test

1. Fix the version in `package.json` by setting it to the latest published version (`22.8.2`, not 20.x!)
2. Check out this branch
3. Install it into any MFE that uses `ModalDialog` using `npm install --save ../paragon`
4. Check any usages of `<ModalDialog>` in `.tsx` files or in `.jsx` files that have `// @ts-check` at the beginning. You can add `// @ts-check` to the .jsx files if it's not there already.

## Merge Checklist


## Post-merge Checklist
